### PR TITLE
Address Selenium warnings

### PR DESCRIPTION
**Pull request by michplunkett**
## Screenshots (if appropriate)
Before change:
Screenshot 2024-11-22 at 3 06 52 PM

After change (selenium warnings are gone:
Screenshot 2024-11-22 at 5 12 41 PM

## Tests and Linting
- [x] This branch is up-to-date with the `develop` branch.
- [x] `pytest` passes on my local development environment.
- [x] `pre-commit` passes on my local development environment.

### CODE REVIEW

File: OpenOversight/tests/conftest.py [hunk 1]
```diff
@@ -906,7 +907,7 @@ def browser(app, server_port):
     # start headless webdriver
     visual_display = Xvfb()
     visual_display.start()
-    driver = webdriver.Firefox(service_log_path="/tmp/geckodriver.log")
+    driver = webdriver.Firefox(service=Service(service_log_path="/tmp/geckodriver.log"))
```
Comment (michplunkett) [inline OpenOversight/tests/conftest.py:910]: https://stackoverflow.com/a/76434055

File: OpenOversight/tests/test_functional.py [hunk 2]
```diff
@@ -163,7 +163,7 @@ def test_find_officer_can_see_uii_question_for_depts_with_uiis(
     ).first()
 
     dept_selector = Select(browser.find_element_by_id("dept"))
-    uii_element = browser.find_element_by_id("uii-question")
+    uii_element = browser.find_element("id", "uii-question")
```
Comment (michplunkett) [inline OpenOversight/tests/test_functional.py:166]: Without this change, I keep getting errors on the 3.12 tests.

File: dockerfiles/web/Dockerfile-test [hunk 3]
```diff
@@ -14,8 +14,8 @@ RUN apt-get update && apt-get install -y xvfb firefox-esr libpq-dev python3-dev
     apt-get install -y libsqlite3-0 && apt-get clean
 
 # install geckodriver
-ENV GECKODRIVER_VERSION="v0.26.0"
-ENV GECKODRIVER_SHA=d59ca434d8e41ec1e30dd7707b0c95171dd6d16056fb6db9c978449ad8b93cc0
+ENV GECKODRIVER_VERSION="v0.35.0"
+ENV GECKODRIVER_SHA=e1b2c3d4e5f67890123456789abcdef0123456789abcdef0123456789abcdef0
```
Comment (michplunkett) [inline dockerfiles/web/Dockerfile-test:18]: Screenshot 2024-11-22 at 3 46 14 PM

File: dockerfiles/web/Dockerfile-test [hunk 4]
```diff
@@ -14,8 +14,8 @@ RUN apt-get update && apt-get install -y xvfb firefox-esr libpq-dev python3-dev
     apt-get install -y libsqlite3-0 && apt-get clean
 
 # install geckodriver
-ENV GECKODRIVER_VERSION="v0.26.0"
-ENV GECKODRIVER_SHA=d59ca434d8e41ec1e30dd7707b0c95171dd6d16056fb6db9c978449ad8b93cc0
+ENV GECKODRIVER_VERSION="v0.35.0"
+ENV GECKODRIVER_SHA=ac26e9ba8f3b8ce0fbf7339b9c9020192f6dcfcbf04a2bcd2af80dfe6bb24260
```
Comment (michplunkett) [inline dockerfiles/web/Dockerfile-test:18]: Screenshot 2024-11-22 at 3 46 14 PM

Driver: [link](https://github.com/mozilla/geckodriver/releases/download/v0.35.0/geckodriver-v0.35.0-linux64.tar.gz)

```console
michaelp@MacBook-Air-18 Downloads % shasum -a 256 geckodriver-v0.35.0-linux64.tar.gz
ac26e9ba8f3b8ce0fbf7339b9c9020192f6dcfcbf04a2bcd2af80dfe6bb24260  geckodriver-v0.35.0-linux64.tar.gz
michaelp@MacBook-Air-18 Downloads %
```

File: requirements.txt [hunk 5]
```diff
@@ -56,13 +56,12 @@ recommonmark==0.7.1
 requests~=2.31.0
 rich~=13.4.2
 s3transfer~=0.6.1
-selenium~=4.10.0
```
Comment (michplunkett) [inline requirements.txt:59]: This is only used in tests, so I moved it to `dev-requirements.txt`.

File: OpenOversight/tests/test_functional.py [hunk 6]
```diff
@@ -230,7 +230,7 @@ def test_incident_detail_do_not_display_read_more_button_for_descriptions_under_
     browser.get(f"http://localhost:{server_port}/officers/1")
 
     # Select incident for officer that has description under cutoff chars
-    result = browser.find_element_by_id("description-overflow-row_1")
+    result = browser.find_element(By.ID, "description-overflow-row_1")
```
Comment (michplunkett) [inline OpenOversight/tests/test_functional.py:233]: Making `find_element` syntax consistent.

**Review by b-meson — APPROVED** [review]
It looks good to me. Are we not worried about the warning in lines  `3700-3701` in your 
"after" screenshot?

**Comment by michplunkett** [comment]
That's a different warning. That can be addressed in another PR, imo.

**Pull request merged**